### DISCUSSION TITLE
Draw lines while moving the mouse

### DISCUSSION THREAD

**Discussion by liucan233** [pixijs/pixijs#8292] [Q&A]
### Expected Behavior

Draw lines while the mouse moves, not the mouse up.
![normal](https://user-images.githubusercontent.com/13808186/163701045-2f505b2c-1eb3-4436-af02-435ca43875c2.jpg)

### Current Behavior
The following code **can work** normally when the **mouse up**.
```js
app.stage.on('mousedown', e => {
    let { x, y } = e.data.global;
    let path = new Graphics();
    path.lineStyle({
        width: 3,
        color: 0x0,
    })
    path.moveTo(x, y);
    // app.stage.addChild(path);

    app.stage.on('mousemove', e => {
        let { x, y } = e.data.global;
        x=x|0;  y=y|0;
        path.lineTo(x, y);
    });

    app.stage.once('mouseup', e => {
        app.stage.removeListener('mousemove');
        app.stage.addChild(path);   // If I do not add path here, I will not be able to draw lines normally.
    });
});
```
The following code **can not work normally** when the mouse move, as shown in the figure below.
```js
app.stage.on('mousedown', e => {
    let { x, y } = e.data.global;
    let path = new Graphics();
    path.lineStyle({
        width: 3,
        color: 0x0,
    })
    path.moveTo(x, y);
    app.stage.addChild(path);

    app.stage.on('mousemove', e => {
        let { x, y } = e.data.global;
        x=x|0;  y=y|0;
        path.lineTo(x, y);
    });

    app.stage.once('mouseup', e => {
        app.stage.removeListener('mousemove');
        // app.stage.addChild(path);
    });
});
```
![error](https://user-images.githubusercontent.com/13808186/163701092-b35e62e4-a026-4429-8c05-7f643f3db053.jpg)

### Possible Solution
None.

### Steps to Reproduce
None.

### Environment

- **`pixi.js` version**: 6.3.0 webgl2
- **Browser & Version**: Microsoft Edge 100.0.1185.44
- **OS & Version**: win10
- **Running Example**: none

**Comment by dev7355608** [comment]
The path is finished before it is rendered. So in the next frame you cannot continue your path where you left off; instead you start a new path. Because `moveTo` was never called for this new path, the origin of the path is set to (0, 0).

You should record the points in an array and clear the graphics and redraw the path after a new point is added.

**Reply by liucan233** [reply]
> The path is finished before it is rendered. So in the next frame you cannot continue your path where you left off; instead you start a new path. Because `moveTo` was never called for this new path, the origin of the path is set to (0, 0).
> 
> You should record the points in an array and clear the graphics and redraw the path after a new point is added.

Thank you for your answer. I need to draw with the mouse. Will this affect the performance?

**Reply by dev7355608** [reply]
Depends on the number of points of course. In your code above you add a point each `mousemove`, which is unnecessary and not even desirable, because it makes the line really jagged.
